src/include/storage/data_table.h

```diff
@@ -224,7 +218,10 @@ class DataTable {
    *
    * @return one past the last tuple slot contained in the data table.
    */
-  SlotIterator end() const;  // NOLINT for STL name compatibility
+
+  SlotIterator end() const {  // NOLINT for STL name compability
+    return SlotIterator();
```

Comment:
Changing this should allow us to remove the constructor without a `DataTable` pointer.
```suggestion
    return InvalidTupleSlot();
```

Reply:
Im not sure this works. i dont think it type checks?

Reply:
Ok, now I get the issue (I was type confuesd...).  I'm fine with this code if you add a comment above the `SlotIterator() = default;` explaining that it is only for constructing an "end" iterator.

Reply:
done